### PR TITLE
Implement IsTextPredictionEnabled in IEntry Handlers

### PR DESCRIPTION
**Pull request by almirvuk**
Implements #381 

- Adds bool IsTextPredictionEnabled { get; } to the IEntry interface
- Adds IsTextPredictionEnabled property map to EntryHandler
- Adds IsTextPredictionEnabled mapping methods to EntryHandler for Android and iOS
- Adds extension methods to apply IsTextPredictionEnabled on Android/iOS
- Adds DeviceTests for initial IsTextPredictionEnabled values on iOS and Android

### CODE REVIEW

**Review by hartez — APPROVED** [review]
Looks good. Thanks!

**Review by rmarinho — APPROVED** [review]

File: src/Core/src/Platform/Android/EntryExtensions.cs
Comment (mattleibow) [inline src/Core/src/Platform/Android/EntryExtensions.cs:55]: I am thinking this is unnecessary setting of the native property. Even though it is fast and, maybe we can use a variable and pipe them all. Then at the end, we set it?

File: src/Core/src/Platform/iOS/EntryExtensions.cs
Comment (mattleibow) [inline src/Core/src/Platform/iOS/EntryExtensions.cs:45]: Is there a way to unset?

**Review by mattleibow — COMMENTED** [review]
Thanks for the PR. Looking good and I just have 2 thoughts.


Not for this PR, but a note for future me and other peoples. We need to create a helper method to test changes. Like init to X, test, set to Y, and confirm it changes. And then do the reverse order.

This is very useful to determine if the "undo" operation works. Like set IsPassword to false, test, true, test and then false again, test.

File: src/Core/src/Platform/iOS/EntryExtensions.cs
Comment (mattleibow) [reply src/Core/src/Platform/iOS/EntryExtensions.cs:45]: To tests this, you can merge this branch into your branch (if we don't merge my PR first) and write a change test like this: https://github.com/dotnet/maui/pull/426/files#diff-5299bd70e5171c15f386cf349244c62764e6373fb1bbcbe0047e901e2b7ebe7cR60

**Review by mattleibow — CHANGES_REQUESTED** [review]
Marking this red just in case I am right and we can't unset the value.

**Comment by almirvuk** [comment]
@mattleibow can you please take a look now? :)

File: src/Core/src/Platform/iOS/EntryExtensions.cs
Comment (almirvuk) [reply src/Core/src/Platform/iOS/EntryExtensions.cs:46]: The last commit should fix this :)

**Comment by mattleibow** [comment]
@almirvuk  could you add this test just so we don't ever break this feature in the future:

```csharp


		[Theory(DisplayName = "IsTextPredictionEnabled Updates Correctly")]
		[InlineData(true, true)]
		[InlineData(true, false)]
		[InlineData(false, true)]
		[InlineData(false, false)]
		public async Task IsTextPredictionEnabledUpdatesCorrectly(bool setValue, bool unsetValue)
		{
			var entry = new EntryStub();

			await ValidatePropertyUpdatesValue(
				entry,
				nameof(IEntry.IsTextPredictionEnabled),
				GetNativeIsTextPredictionEnabled,
				setValue,
				unsetValue);
		}
```

File: src/Core/src/Platform/iOS/EntryExtensions.cs
Comment (mattleibow) [reply src/Core/src/Platform/iOS/EntryExtensions.cs:46]: Awesome!

**Review by mattleibow — APPROVED** [review]
Looking great! Besides the test that I think we should add, this is good to merge!

Thanks for the PR and the work.

**Comment by almirvuk** [comment]
> @almirvuk could you add this test just so we don't ever break this feature in the future:
> 
> ```cs
> 		[Theory(DisplayName = "IsTextPredictionEnabled Updates Correctly")]
> 		[InlineData(true, true)]
> 		[InlineData(true, false)]
> 		[InlineData(false, true)]
> 		[InlineData(false, false)]
> 		public async Task IsTextPredictionEnabledUpdatesCorrectly(bool setValue, bool unsetValue)
> 		{
> 			var entry = new EntryStub();
> 
> 			await ValidatePropertyUpdatesValue(
> 				entry,
> 				nameof(IEntry.IsTextPredictionEnabled),
> 				GetNativeIsTextPredictionEnabled,
> 				setValue,
> 				unsetValue);
> 		}
> ```

Added with the last commit :) Thank you @mattleibow

**Comment by mattleibow** [comment]
Absolutely stunning! This needs to be merged as soon as CI is green.

File: src/Core/src/Platform/Android/EntryExtensions.cs
Comment (bkaankose) [reply src/Core/src/Platform/Android/EntryExtensions.cs:55]: This enhancement is addressed in #507.

**Pull request merged**
